src/Settings/ProjectSettings.ts

```diff
@@ -23,10 +24,30 @@ export class ProjectSettings extends Settings {
     }
 
     /**
-     * Gets a value by key from Azure DevOps.
+     * Gets the value of the given key from the given map of existing Azure DevOps project settings. If the key
+     * does not exist, then the given default value is returned.
      */
-    public get(key: string): PromiseLike<string> {
-        return super.get(this.getProjectSpecificKey(key));
+    public getOrDefault(key: string, existingSettings : Map<string, string>, defaultValue: string): string {
+        const settingKey = this.getProjectSpecificKey(key);
+        if(existingSettings.has(settingKey)){
+            return existingSettings.get(settingKey);
+        }
+        return defaultValue;
+    }
+
+    /** Fetches and returns the stored project settings. */
+    public async loadStoredProjectSettings(): Promise<Map<string, string>> {
```

Comment:
question: Do we really need to expose this to the caller? Maybe we can have a `storedSettingsMap` within the `ProjectSettings`/`Settings` object and call this lazily in `getOrDefault`/`get` (if `storedSettingsMap` has not been set yet)?

Reply:
Great idea! Implemented your suggestion and now the settings are lazily loaded whenever needed in `Settings.ts`